src/object-pool.ts

```diff
@@ -1,10 +1,287 @@
-import {ArmorObjectPoolOptions} from './object-pool-options';
-import {ArmorObjectPoolState} from './object-pool-state';
+import ArmorCollection from './collection';
+import ArmorCollectionSelector from './selector';
+import ArmorObjectPoolInstance from './object-pool-instance';
+import ArmorObjectPoolOptions from './object-pool-options';
+import ArmorObjectPoolState from './object-pool-state';
 
-export class ArmorObjectPool<T> {
-	public readonly state: ArmorObjectPoolState;
+export default class ArmorObjectPool<T> implements ArmorCollection<T> {
+	public state: ArmorObjectPoolState<T>;
+	public startSize: number = 10;
+	public readonly poolObj: ArmorObjectPoolInstance<T>;
 
-	constructor(options?: ArmorObjectPoolOptions) {
-		this.state = new ArmorObjectPoolState(options);
+	constructor(poolObj: ArmorObjectPoolInstance<T>, options?: ArmorObjectPoolOptions<T>) {
+		if (typeof poolObj !== 'function') {
+			throw new Error('Must have a comparator function for priority queue to operate properly');
+		}
+		this.poolObj = poolObj;
+
+		this.state = {
+			type: 'opState',
+			elements: [],
+			objectCount: 0,
+			maxSize: 1000,
+			autoIncrease: false,
+			increaseFactor: 2,
+			increaseBreakPoint: 0.8
+		};
+
+		if (options) {
+			this.parseOptions(options);
+		}
+
+		this.increase(this.startSize);
+	}
+	public parseOptions(options: ArmorObjectPoolOptions<T>): void {
```

Comment:
Per the readonly state comment above, testing is a little cleaner and easier when parse functions return a value rather than set values in the current class. To make this work you'll need a defined type to return for the state, and then changes at the top of each parse function.
Start by defining the default object at the top of the function, then do a check for the required options argument and return the default if its missing or invalid. If it's not missing, fill out the options as you parse them.